{
  Throwable error=null;
  try {
    output.process(exchange);
    if (exchange.getException() != null) {
      error=exchange.getException();
    }
  }
 catch (  Throwable e) {
    error=e;
  }
  if (error != null) {
    if (!customProcessorForException(exchange,error)) {
      logError(exchange,error);
    }
  }
}
